internal/codegen/rest_method.go

```diff
@@ -32,6 +36,11 @@ func (m *Method) restMethodFuncReturnParams(def *Group) {
 	case protocol.Method_get:
 		def.Add(m.Return.PointerType())
 		def.Error()
+	case protocol.Method_create:
+		if m.EntityPathKey != nil && m.EntityPathKey.Type.Primitive != nil {
```

Comment:
Is `EntityPathKey` ever expected to be nil? Shouldn't create always have one?

Also does that mean you are explicitly _not_ supporting non-primitive resource types for Create? In which case can you add a log message to that effect and skip creating the method if so? Note that you should still support primitive-backed typerefs

Reply:
> s EntityPathKey ever expected to be nil? Shouldn't create always have one?

I wasn't clear if it can be nil for something like a Simple type. There was nothing in the test suite I could reference for that case since the simple defined there only has get/update/delete methods. 

> Also does that mean you are explicitly not supporting non-primitive resource types for Create? In which case can you add a log message to that effect and skip creating the method if so? Note that you should still support primitive-backed typerefs

Actually it is supported for other types and it _should_ generate a valid/usable method, it just won't expose the ID of the object you created (in many cases that is fine since you know it at creation time)

Reply:
https://linkedin.github.io/rest.li/spec/protocol#simple-resources
Simple resources don't have CREATE, so that's not going to be a problem